web-common/src/features/dashboards/state-managers/loaders/DashboardStateDataLoader.ts

```diff
@@ -247,38 +280,57 @@ export class DashboardStateDataLoader {
       return undefined;
     const metricsViewSpec = validSpecResp.data.metricsView;
     const exploreSpec = validSpecResp.data.explore;
+    const rillDefaultExploreState = get(this.rillDefaultExploreState);
+    const exploreStateFromYAMLConfig = get(this.exploreStateFromYAMLConfig);
     const explorePresetFromYAMLConfig = get(this.explorePresetFromYAMLConfig);
-    if (!explorePresetFromYAMLConfig.data) return undefined;
+    if (
+      !rillDefaultExploreState?.data ||
+      !explorePresetFromYAMLConfig?.data ||
+      !exploreStateFromYAMLConfig?.data
+    ) {
+      return undefined;
+    }
 
     // Pressing back button and going back to empty url state should not restore from session store
     const backButtonUsed = type === "popstate";
     const skipSessionStorage = backButtonUsed;
 
-    const { partialExploreState: partialExploreStateFromUrl } =
-      convertURLSearchParamsToExploreState(
+    const exploreStateFromSessionStorage =
+      getPartialExploreStateFromSessionStorage(
+        this.exploreName,
+        this.storageNamespacePrefix,
         urlSearchParams,
         metricsViewSpec,
         exploreSpec,
-        explorePresetFromYAMLConfig.data,
       );
-    // If we are skipping using state from session storage then exit early with partialExploreStateFromUrl
-    // regardless if there is exploreStateFromSessionStorage for current url params or not.
-    if (skipSessionStorage) return partialExploreStateFromUrl;
 
-    const exploreStateFromSessionStorage =
-      getPartialExploreStateFromSessionStorage(
-        this.exploreName,
-        this.storageNamespacePrefix,
+    const { partialExploreState: partialExploreStateFromUrl } =
+      convertURLSearchParamsToExploreState(
         urlSearchParams,
         metricsViewSpec,
         exploreSpec,
+        explorePresetFromYAMLConfig.data,
       );
 
-    return (
-      // preference goes to session storage 1st.
-      exploreStateFromSessionStorage ??
-      // else we use the partial explore state from the url params.
-      partialExploreStateFromUrl
-    );
+    const exploreStateOrder = [
```

Comment:
Why is this and the `cascadingExploreStateMerge` repeated twice in this file?

Reply:
The items in the order are slightly different for init state and state for every url.